server/src/main/java/org/apache/cassandra/sidecar/restore/RestoreProcessor.java

```diff
@@ -170,9 +169,6 @@ public void execute(Promise<Void> promise)
                 break;
             }
 
-            Preconditions.checkState(range.canProduceTask(),
```

Comment:
do we not need this check anymore?

Reply:
It is checked in `org.apache.cassandra.sidecar.db.RestoreRange#toAsyncTask` already. The check here is redundant and disruptive.